build-system/tasks/dist.js

```diff
@@ -146,10 +146,17 @@ async function doDist(extraArgs = {}) {
   }
   await buildExtensions(options);
 
-  if (!argv.core_runtime_only) {
+  // Steps that are to be run only during a full `amp dist`.
+  if (
+    !argv.core_runtime_only &&
+    !argv.extensions &&
+    !argv.extensions_from &&
+    !argv.noextensions
+  ) {
     await buildVendorConfigs(options);
     await formatExtractedMessages();
```

Comment:
Any binary can reference its output, so you need to `formatExtractMessages` on every build.

Reply:
Gotcha. Updated.

Reply:
`core_runtime_only` needs it too.

Reply:
This hasn't been the case for the past couple of years. IIRC, @jridgewell explicitly requested that we do nothing other than compile `v0.js` when that flag is used. Has this changed?

Reply:
`v0.js` can reference the table, so it would become unusable once extracted log messages is turned on.

Reply:
Aha, the feature is as-yet unused. Updated again.